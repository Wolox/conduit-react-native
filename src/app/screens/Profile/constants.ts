const ANDROID_KEYBOARD_OFFSET = 10;
export const ANDROID_SCROLLVIEW_PROPS = {
  enableOnAndroid: true,
  extraScrollHeight: ANDROID_KEYBOARD_OFFSET,
  extraHeight: ANDROID_KEYBOARD_OFFSET
};
