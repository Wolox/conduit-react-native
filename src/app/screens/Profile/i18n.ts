import i18next from 'i18next';

i18next.addResources('en', 'PROFILE', {
  LOGOUT: 'Logout',
  USERNAME: 'Username',
  EMAIL: 'E-mail',
  DESCRIPTION: 'Description about you',
  UPDATE_PROFILE: 'Update profile'
});
