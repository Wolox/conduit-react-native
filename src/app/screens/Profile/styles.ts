import { StyleSheet } from 'react-native';
import { transparent, green, white } from '@constants/colors';

const COMMON_VALUE_25 = 25;
const SMALL_SPACING = 10;
const SPACING = 30;
const COMMON_VALUE_5 = 5;

const FORM_ELEMENT_COMMON_PROPS = {
  margin: COMMON_VALUE_5,
  backgroundColor: transparent,
  height: SPACING
};

export default StyleSheet.create({
  container: {
    padding: 16,
    paddingTop: 0,
    flex: 1,
    paddingBottom: COMMON_VALUE_5,
    justifyContent: 'space-between'
  },
  containerData: {
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center'
  },
  form: {
    paddingTop: SPACING,
    paddingHorizontal: SMALL_SPACING
  },
  stretchAndFlex: {
    alignSelf: 'stretch'
  },
  formElement: {
    ...FORM_ELEMENT_COMMON_PROPS,
    marginBottom: SPACING
  },
  firstElement: {
    marginBottom: 45
  },
  lastElement: {
    marginBottom: SMALL_SPACING
  },
  formButton: {
    backgroundColor: green,
    padding: 10,
    borderRadius: COMMON_VALUE_25,
    marginTop: 2
  },
  textFormButton: {
    color: white,
    fontWeight: 'bold',
    width: '60%',
    textAlign: 'center'
  },
  textFormCustomText: {
    marginTop: COMMON_VALUE_25
  }
});
