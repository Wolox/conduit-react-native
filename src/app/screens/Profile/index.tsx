import React, { useEffect, useState } from 'react';
import { Keyboard, ScrollView, TouchableOpacity, View } from 'react-native';
import i18next from 'i18next';
import { useDispatch, useSelector } from 'react-redux';
import { useForm } from 'react-hook-form';
import WithHeader from '@components/WithHeader';
import CustomButton from '@components/CustomButton';
import ControlledCustomTextInput from '@components/CustomTextInput/controller';
import CustomText from '@components/CustomText';
import Routes from '@constants/routes';
import { isIos } from '@constants/platform';
import { OPACITY } from '@constants/commonStyles';
import { Navigation } from '@interfaces/navigation';
import { actionCreators as AuthActions } from '@redux/auth/actions';
import { actionCreators as FeedbackActions } from '@redux/feedback/actions';
import logoutIcon from '@assets/Profile/icLogout.png';
import { validateRequired, validateEmail, validateOnlyText } from '@utils/validations/validateUtils';
import signUpStyles from '@screens/Auth/screens/SignUp/styles';
import { State } from '@interfaces/reduxInterfaces';
import { CurrentUser } from '@interfaces/authInterfaces';
import CustomModal from '@components/CustomModal';
import { getAvatar } from '@constants/iconsConstants';

import './i18n';

import styles from './styles';
import ProfileListItem from './components/ProfileItem';
import { ANDROID_SCROLLVIEW_PROPS, ProfileFormValues, FIELDS } from './constants';
import AvatarSelector from './components/AvatarSelector';

interface Props extends Navigation {}

function Profile() {
  const dispatch = useDispatch();
  const handlePressLogout = () => dispatch(AuthActions.logout());
  const currentUser = useSelector((state: State) => state.auth.currentUser?.user);
  const { userProfileLoading, currentUserLoading, currentUserError } = useSelector(
    (state: State) => state.auth
  );
  const [usernameError, setUsernameError] = useState('');
  const [emailError, setEmailError] = useState('');
  const {
    handleSubmit,
    control,
    formState: { isValid, isDirty }
  } = useForm<ProfileFormValues>({ mode: 'onBlur' });

  const usernameExist = () => setUsernameError(i18next.t('PROFILE:USERNAME_EXIST'));

  const emailExist = () => setEmailError(i18next.t('PROFILE:EMAIL_EXIST'));
  const hasError = !!currentUserError;

  const [avatarSelected, setAvatarSelected] = useState(currentUser?.image || '');
  const [isAvatarChanged, setIsAvatarChanged] = useState(false);
  const handleUpdateProfile = (values: ProfileFormValues) => {
    Keyboard.dismiss();
    const user: CurrentUser = {
      email: values.email,
      token: currentUser?.token,
      username: values.username,
      bio: values.description,
      image: avatarSelected
    };
    if (currentUser?.username === values.username) {
      dispatch(AuthActions.updateCurrentUser({ user }, emailExist));
    } else {
      dispatch(AuthActions.getUserProfile({ user }, usernameExist, emailExist));
    }
  };
  const onEmailFocus = () => setEmailError('');
  const onUsenameFocus = () => setUsernameError('');
  const handleChangeAvatar = () =>
    dispatch(
      FeedbackActions.showModal(
        <CustomModal
          title={i18next.t('PROFILE:SELECT_AVATAR')}
          body={<AvatarSelector selected={avatarSelected} setSelected={setAvatarSelected} />}
          style={styles.modalContainer}
        />
      )
    );
  useEffect(() => {
    setIsAvatarChanged(avatarSelected !== currentUser?.image);
  }, [avatarSelected, currentUser]);
  return (
    <WithHeader
      title={i18next.t(`app:${Routes.Profile}`)}
      withAvatar
      avatar={getAvatar(avatarSelected)}
      onPressAvatar={handleChangeAvatar}>
      <View style={styles.container}>
        <ScrollView
          bounces={false}
          contentContainerStyle={styles.containerData}
          keyboardShouldPersistTaps="handled"
          showsVerticalScrollIndicator={false}
          style={styles.stretchAndFlex}
          {...(isIos ? {} : ANDROID_SCROLLVIEW_PROPS)}>
          <TouchableOpacity
            onPress={Keyboard.dismiss}
            activeOpacity={OPACITY.NONE}
            style={[styles.stretchAndFlex, styles.form]}>
            <ControlledCustomTextInput
              name={FIELDS.username}
              control={control}
              label={i18next.t('PROFILE:USERNAME')}
              placeholder={i18next.t('PROFILE:USERNAME')}
              inputContainerStyle={[signUpStyles.input, styles.inputCustom]}
              labelStyle={signUpStyles.labelText}
              errorContainerStyle={signUpStyles.errorContainer}
              showError={hasError}
              rules={{ ...validateRequired, ...validateOnlyText }}
              defaultValue={currentUser?.username}
              onFocus={onUsenameFocus}
            />
            <CustomText error xsmall style={styles.otherErrorStyle}>
              {usernameError}
            </CustomText>
            <ControlledCustomTextInput
              name={FIELDS.email}
              control={control}
              label={i18next.t('PROFILE:EMAIL')}
              placeholder={i18next.t('PROFILE:EMAIL')}
              inputContainerStyle={[signUpStyles.input, styles.inputCustom]}
              labelStyle={signUpStyles.labelText}
              errorContainerStyle={signUpStyles.errorContainer}
              showError={hasError}
              rules={{ ...validateRequired, ...validateEmail }}
              defaultValue={currentUser?.email}
              onFocus={onEmailFocus}
            />
            <CustomText error xsmall style={styles.otherErrorStyle}>
              {emailError}
            </CustomText>
            <ControlledCustomTextInput
              name={FIELDS.description}
              control={control}
              label={i18next.t('PROFILE:DESCRIPTION')}
              placeholder={i18next.t('PROFILE:DESCRIPTION')}
              inputContainerStyle={[signUpStyles.input, styles.inputCustom]}
              labelStyle={signUpStyles.labelText}
              errorContainerStyle={signUpStyles.errorContainer}
              showError={hasError}
              rules={{ ...validateOnlyText }}
              defaultValue={currentUser?.bio}
            />
            <CustomButton
              disabled={!isAvatarChanged && !(isValid && isDirty)}
              onPress={handleSubmit(handleUpdateProfile)}
              style={styles.formButton}
              textStyle={styles.textFormButton}
              title={i18next.t('PROFILE:UPDATE_PROFILE')}
              loading={userProfileLoading || currentUserLoading}
            />
          </TouchableOpacity>
          <ProfileListItem
            title={i18next.t('PROFILE:LOGOUT')}
            icon={logoutIcon}
            onPress={handlePressLogout}
          />
        </ScrollView>
      </View>
    </WithHeader>
  );
}

export default Profile;
