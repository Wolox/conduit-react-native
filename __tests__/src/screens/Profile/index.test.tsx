import React from 'react';
import renderer from 'react-test-renderer';
import * as redux from 'react-redux';
import Profile from '@screens/Profile';

jest.mock('react-redux', () => {
  return {
    useSelector: jest.fn().mockImplementation(() => ({ currentUser: '' })),
    useDispatch: () => jest.fn()
  };
});

describe('Profile', () => {
  beforeEach(() => {
    const useSelectorMock = jest.spyOn(redux, 'useSelector');
    useSelectorMock.mockReturnValue({
      userProfileLoading: false,
      currentUserLoading: false,
      currentUserError: null
    });
    jest.useFakeTimers();
  });
  test('Profile Snapshot', () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).not.toBeNull();
  });
});
