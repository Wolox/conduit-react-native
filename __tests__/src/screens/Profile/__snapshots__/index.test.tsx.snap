// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Profile Snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#5CB85C",
        "width": "100%",
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "minHeight": 168.14285714285714,
            "paddingHorizontal": 16,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 2,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontFamily": "Nunito",
                  "fontSize": 25.142857142857142,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                },
                Array [
                  Object {
                    "color": "#FFF",
                    "fontFamily": "Nunito",
                    "fontSize": 31.428571428571427,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                  },
                ],
                Object {
                  "marginBottom": 5,
                },
              ]
            }
          >
            app:Profile
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>,
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 14,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#5CB85C",
              "flex": 1,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
            "borderTopLeftRadius": 14,
            "borderTopRightRadius": 14,
            "height": "100%",
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
          Object {
            "paddingTop": 97.14285714285714,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
            "padding": 16,
            "paddingBottom": 5,
            "paddingTop": 0,
          }
        }
      >
        <RCTScrollView
          bounces={false}
          contentContainerStyle={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          keyboardShouldPersistTaps="handled"
          showsVerticalScrollIndicator={false}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
        >
          <View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 30,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "marginBottom": 5,
                      "padding": 10,
                    },
                    false,
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000",
                        "fontFamily": "Nunito",
                        "fontSize": 25.142857142857142,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Array [
                        Object {
                          "color": "#A7A7A7",
                        },
                        Object {
                          "fontSize": 22,
                        },
                      ],
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#000",
                          "fontFamily": "Nunito",
                          "fontSize": 12,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "marginTop": 5,
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "#5CB85C",
                          "paddingLeft": 5,
                          "textAlign": "left",
                          "width": "100%",
                        },
                      ],
                    ]
                  }
                >
                  PROFILE:USERNAME
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "backgroundColor": "#FFF",
                        "borderBottomWidth": 0,
                        "borderColor": "#555555",
                        "borderRadius": 15,
                        "elevation": 1,
                        "flexDirection": "row",
                        "height": 50,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 8,
                        "shadowColor": "#E5E5E5",
                        "shadowOffset": Object {
                          "height": 3,
                          "width": 0,
                        },
                        "shadowOpacity": 1,
                      },
                      Object {},
                      Array [
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#FFF",
                          "borderBottomWidth": 0,
                          "borderRadius": 15,
                          "elevation": 8,
                          "height": 40,
                          "paddingHorizontal": 8,
                          "shadowColor": "#E5E5E5",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 1,
                          "width": "100%",
                        },
                        Object {
                          "borderColor": "#F1F1F1",
                          "borderWidth": 1,
                        },
                      ],
                      Array [
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#FFF",
                          "borderBottomWidth": 0,
                          "borderRadius": 15,
                          "elevation": 8,
                          "height": 40,
                          "paddingHorizontal": 8,
                          "shadowColor": "#E5E5E5",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 1,
                          "width": "100%",
                        },
                        Object {
                          "borderColor": "#F1F1F1",
                          "borderWidth": 1,
                        },
                      ],
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={false}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    clearButtonMode="never"
                    editable={true}
                    keyboardType="default"
                    multiline={false}
                    name="username"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="PROFILE:USERNAME"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    style={
                      Array [
                        Object {
                          "color": "#555555",
                          "fontFamily": "Nunito",
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "margin": 0,
                          "padding": 0,
                        },
                        Object {
                          "flex": 1,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": 15,
                        "marginTop": 3,
                      },
                      Object {
                        "alignSelf": "center",
                        "paddingLeft": 5,
                        "textAlign": "left",
                        "width": "100%",
                      },
                    ]
                  }
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000",
                      "fontFamily": "Nunito",
                      "fontSize": 25.142857142857142,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "fontSize": 18.857142857142858,
                      },
                      Object {
                        "color": "#FF0000",
                      },
                    ],
                    Object {
                      "marginTop": -20,
                      "paddingLeft": 20,
                    },
                  ]
                }
              >
                
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "marginBottom": 5,
                      "padding": 10,
                    },
                    false,
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000",
                        "fontFamily": "Nunito",
                        "fontSize": 25.142857142857142,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Array [
                        Object {
                          "color": "#A7A7A7",
                        },
                        Object {
                          "fontSize": 22,
                        },
                      ],
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#000",
                          "fontFamily": "Nunito",
                          "fontSize": 12,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "marginTop": 5,
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "#5CB85C",
                          "paddingLeft": 5,
                          "textAlign": "left",
                          "width": "100%",
                        },
                      ],
                    ]
                  }
                >
                  PROFILE:EMAIL
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "backgroundColor": "#FFF",
                        "borderBottomWidth": 0,
                        "borderColor": "#555555",
                        "borderRadius": 15,
                        "elevation": 1,
                        "flexDirection": "row",
                        "height": 50,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 8,
                        "shadowColor": "#E5E5E5",
                        "shadowOffset": Object {
                          "height": 3,
                          "width": 0,
                        },
                        "shadowOpacity": 1,
                      },
                      Object {},
                      Array [
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#FFF",
                          "borderBottomWidth": 0,
                          "borderRadius": 15,
                          "elevation": 8,
                          "height": 40,
                          "paddingHorizontal": 8,
                          "shadowColor": "#E5E5E5",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 1,
                          "width": "100%",
                        },
                        Object {
                          "borderColor": "#F1F1F1",
                          "borderWidth": 1,
                        },
                      ],
                      Array [
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#FFF",
                          "borderBottomWidth": 0,
                          "borderRadius": 15,
                          "elevation": 8,
                          "height": 40,
                          "paddingHorizontal": 8,
                          "shadowColor": "#E5E5E5",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 1,
                          "width": "100%",
                        },
                        Object {
                          "borderColor": "#F1F1F1",
                          "borderWidth": 1,
                        },
                      ],
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={false}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    clearButtonMode="never"
                    editable={true}
                    keyboardType="default"
                    multiline={false}
                    name="email"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="PROFILE:EMAIL"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    style={
                      Array [
                        Object {
                          "color": "#555555",
                          "fontFamily": "Nunito",
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "margin": 0,
                          "padding": 0,
                        },
                        Object {
                          "flex": 1,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": 15,
                        "marginTop": 3,
                      },
                      Object {
                        "alignSelf": "center",
                        "paddingLeft": 5,
                        "textAlign": "left",
                        "width": "100%",
                      },
                    ]
                  }
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000",
                      "fontFamily": "Nunito",
                      "fontSize": 25.142857142857142,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "fontSize": 18.857142857142858,
                      },
                      Object {
                        "color": "#FF0000",
                      },
                    ],
                    Object {
                      "marginTop": -20,
                      "paddingLeft": 20,
                    },
                  ]
                }
              >
                
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "marginBottom": 5,
                      "padding": 10,
                    },
                    false,
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000",
                        "fontFamily": "Nunito",
                        "fontSize": 25.142857142857142,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Array [
                        Object {
                          "color": "#A7A7A7",
                        },
                        Object {
                          "fontSize": 22,
                        },
                      ],
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#000",
                          "fontFamily": "Nunito",
                          "fontSize": 12,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "marginTop": 5,
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "#5CB85C",
                          "paddingLeft": 5,
                          "textAlign": "left",
                          "width": "100%",
                        },
                      ],
                    ]
                  }
                >
                  PROFILE:DESCRIPTION
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "backgroundColor": "#FFF",
                        "borderBottomWidth": 0,
                        "borderColor": "#555555",
                        "borderRadius": 15,
                        "elevation": 1,
                        "flexDirection": "row",
                        "height": 50,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 8,
                        "shadowColor": "#E5E5E5",
                        "shadowOffset": Object {
                          "height": 3,
                          "width": 0,
                        },
                        "shadowOpacity": 1,
                      },
                      Object {},
                      Array [
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#FFF",
                          "borderBottomWidth": 0,
                          "borderRadius": 15,
                          "elevation": 8,
                          "height": 40,
                          "paddingHorizontal": 8,
                          "shadowColor": "#E5E5E5",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 1,
                          "width": "100%",
                        },
                        Object {
                          "borderColor": "#F1F1F1",
                          "borderWidth": 1,
                        },
                      ],
                      Array [
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#FFF",
                          "borderBottomWidth": 0,
                          "borderRadius": 15,
                          "elevation": 8,
                          "height": 40,
                          "paddingHorizontal": 8,
                          "shadowColor": "#E5E5E5",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 1,
                          "width": "100%",
                        },
                        Object {
                          "borderColor": "#F1F1F1",
                          "borderWidth": 1,
                        },
                      ],
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={false}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    clearButtonMode="never"
                    editable={true}
                    keyboardType="default"
                    multiline={false}
                    name="description"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="PROFILE:DESCRIPTION"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    style={
                      Array [
                        Object {
                          "color": "#555555",
                          "fontFamily": "Nunito",
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "margin": 0,
                          "padding": 0,
                        },
                        Object {
                          "flex": 1,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": 15,
                        "marginTop": 3,
                      },
                      Object {
                        "alignSelf": "center",
                        "paddingLeft": 5,
                        "textAlign": "left",
                        "width": "100%",
                      },
                    ]
                  }
                />
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#F1F1F1",
                    "borderColor": "#A7A7A7",
                    "borderRadius": 25,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 2,
                    "opacity": 1,
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000",
                        "fontFamily": "Nunito",
                        "fontSize": 25.142857142857142,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "Nunito",
                          "fontSize": 25.142857142857142,
                          "fontStyle": "normal",
                          "fontWeight": "700",
                        },
                        Object {
                          "color": "#FFF",
                        },
                      ],
                      Array [
                        Array [],
                        Object {
                          "color": "#FFF",
                          "fontWeight": "bold",
                          "textAlign": "center",
                          "width": "60%",
                        },
                        Object {
                          "color": "#939393",
                        },
                      ],
                    ]
                  }
                >
                  PROFILE:UPDATE_PROFILE
                </Text>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "justifyContent": "space-between",
                      "paddingRight": 10,
                      "paddingTop": 20,
                    },
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "paddingVertical": 4,
                      "width": "95%",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "paddingVertical": 4,
                      "width": "95%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "borderColor": "#5CB85C",
                        "borderRadius": 26.785714285714285,
                        "borderWidth": 0.5,
                        "height": 53.57142857142857,
                        "justifyContent": "center",
                        "marginRight": 14,
                        "width": 53.57142857142857,
                      }
                    }
                  >
                    <Image
                      resizeMode="contain"
                      source={
                        Object {
                          "process": [Function],
                        }
                      }
                      style={
                        Object {
                          "height": 49.285714285714285,
                          "tintColor": "#5CB85C",
                          "width": 49.285714285714285,
                        }
                      }
                    />
                  </View>
                  <Text
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#000",
                          "fontFamily": "Nunito",
                          "fontSize": 25.142857142857142,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                        },
                        Array [],
                        undefined,
                      ]
                    }
                  >
                    PROFILE:LOGOUT
                  </Text>
                </View>
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "height": 25.714285714285715,
                      "tintColor": "#E5E5E5",
                      "width": 15,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "position": "absolute",
          "top": -117.14285714285714,
          "width": "100%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "#F1F1F1",
              "borderColor": "#FFF",
              "borderRadius": 107.14285714285714,
              "borderWidth": 0.5,
              "height": 214.28571428571428,
              "width": 214.28571428571428,
            },
            false,
          ]
        }
      />
    </View>
  </View>,
]
`;
